src/Primitives/src/StringValues.cs

```diff
@@ -218,25 +218,41 @@ static string GetJoinedStringValueFromArray(string[] values)
 
         public string[] ToArray()
         {
-            return GetArrayValue() ?? Array.Empty<string>();
+            // The same idea as GetArrayValue with slightly different semantics: This method always returns a copy of the array (just like the IEnumerable.ToArray extension method).
+
+            // Take local copy of _values so type checks remain valid even if the StringValues is overwritten in memory.
+            var value = _values;
+            switch (value)
+            {
+                case string[] values:
+#if NETCOREAPP
+                    return (string[]) values.AsSpan().ToArray();
```

Comment:
Doesn't need the cast
```suggestion
                    return values.AsSpan().ToArray();
```